index.html

```diff
@@ -1028,6 +1061,461 @@ <h2>
+          </td>
+        </tr>
+      </table>
+      <dl data-dfn-for="GamepadHapticActuator">
+        <dt>
+          <dfn>effects</dfn> attribute
+        </dt>
+        <dd>
+          <p>
+            Array of {{Gamepad/GamepadHapticEffectType}} values representing all
+            the types of haptic effects that the actuator supports. This
+            property MUST contain all the {{Gamepad/GamepadHapticEffectType}}
+            values that the actuator supports, unless the [=user agent=] does
+            not support playing effects of that type. The same object MUST be
+            returned and MUST NOT change for the lifetime of the
+            {{GamepadHapticActuator}}.
+          </p>
+          <p>
+            The {{GamepadHapticActuator/effects}} getter steps are:
+          </p>
+          <ol>
+            <li>Return [=this=].{{GamepadHapticActuator/[[effects]]}}.
+            </li>
+          </ol>
+        </dd>
+        <dt>
+          <dfn>playEffect()</dfn> method
+        </dt>
+        <dd>
+          <p>
+            The {{GamepadHapticActuator/playEffect()}} method steps are:
+          </p>
+          <ol>
+            <li>If |params:GamepadEffectParameters| does not describe a [=valid
+            effect=] of type |type:GamepadHapticEffectType|, return [=a promise
+            rejected with=] reason {{TypeError}}.
+            </li>
+            <li>Let |document| be the [=current settings object=]'s [=relevant
+            global object=]'s [=associated `Document`=].
+            </li>
+            <li>If |document| is `null` or |document| is not [=Document/fully
+            active=] or |document|'s [=Document/visibility state=] is `"hidden"`, return
+            [=a promise resolved with=] {{GamepadHapticsResult/"preempted"}}.
+            </li>
+            <li>If [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}
+            is not `undefined`:
+              <ol>
+                <li>Let |effectPromise| be
+                [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}.
+                </li>
+                <li>Set
+                [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}} to
+                `undefined`.
+                </li>
+                <li>[=Queue a global task=] on the [=relevant global object=]
+                of [=this=] using the [=gamepad task source=] to [=resolve=]
+                |effectPromise| with {{GamepadHapticsResult/"preempted"}}.
+                </li>
+              </ol>
+            </li>
+            <li>If this actuator cannot [=play effects with type=] |type|,
+            return [=a promise rejected with=] reason {{NotSupportedError}}.
+            </li>
+            <li>Let {{GamepadHapticActuator/[[playingEffectPromise]]}} be [=a
+            new promise=].
+            </li>
+            <li>Do the following steps in parallel:
+              <ol>
+                <li>[=Issue a haptic effect=] to the actuator with |type| and
```

Comment:
I wonder if we should take a snapshot of a DOMHighResTimestamp before going on parallel, and then sending that along over as well. That would allow for more precise playback timing relative to when this method was called.